alphadia/transferlearning/metrics.py

```diff
@@ -5,67 +5,55 @@
-        Accumulate a metric at a given epoch.
-
-        Parameters
-        ----------
-        epoch : int
-            The epoch at which the metric was calculated.
-        loss : float
-            The value of the metric.
-
-        """
-
-        new_stats = pd.DataFrame({self.name: [loss]})
-        new_stats.index = [epoch]
-        new_stats.index.name = "epoch"
-        self.stats = (
-            new_stats if self.stats is None else pd.concat([self.stats, new_stats])
-        )
-
-
 class TestMetricBase:
     """
-    An Base class for test metrics. Test metrics are classes that calculate a metric on the test set at a given epoch
-    and accumulate the metric over time for reporting.
+    An Base class for test metrics. Test metrics are classes that calculate a metric on the test set at a given epoch.
     """
 
     def __init__(self, columns: List[str]):
         self.columns = columns  # a list of column names for the stats dataframe
-        self.stats = (
-            None  # Stats is a pandas dataframe that stores the test metric over time
-        )
 
-    def _update_stats(self, new_stats: pd.DataFrame, epoch: int):
+    def _to_long_format(
+        self, stats: pd.DataFrame, epoch: int, dataset: str, property_name: str
+    ):
         """
-        Update the stats dataframe with new stats at a given epoch.
+        Convert the stats dataframe to a long format.
 
         Parameters
         ----------
-        new_stats : pd.DataFrame
-            A pandas dataframe containing the new stats.
+        stats : pd.DataFrame
+            A pandas dataframe containing the stats.
         epoch : int
-            The epoch at which the new stats were calculated.
+            The epoch at which the stats were calculated.
+        dataset : str
+            The name of the dataset. e.g. "train", "validation", "test".
+        property_name : str
+            The name of the property. e.g. "charge", "rt"
+
+        Returns
+        -------
+        pd.DataFrame
+            A pandas dataframe containing the stats in a long format.
 
         """
-        new_stats.index = [epoch]
-        new_stats.index.name = "epoch"
-        self.stats = (
-            new_stats if self.stats is None else pd.concat([self.stats, new_stats])
-        )
+        long = []
+        for row in stats.iterrows():
+            epoch = epoch
+            for column in stats.columns:
+                if column != "epoch":
+                    subject = {
+                        "dataset": dataset,
+                        "epoch": epoch,
+                        "property": property_name,
+                        "metric_name": column,
+                        "value": row[1][column],
+                    }
+                    long.append(subject)
+        return long
 
-    def calculate_test_metric(self, test_input: dict, epoch: int):
+    def calculate_test_metric(
+        self, test_input: dict, epoch: int, dataset: str, property_name: str
```

Comment:
Quick question :D We pass a dict because the MS2 metrics need acces to the `psm_df`?